tables/DataMan/DataManager.cc

```diff
@@ -316,7 +317,8 @@ DataManagerCtor DataManager::getCtor (const String& type)
         libname = libname.substr (0, pos);
     }
     // Try to load and initialize the dynamic library.
-    DynLib dl(libname, string("libcasa_"), "register_"+libname, False);
+    DynLib dl(libname, string("libcasa_"), STRING(CASACORE_MAJOR_VERSION),
```

Comment:
Passing it in from cmake probably makes most sense as there the soversion is defined.
If you couple it with the major version it must be bumped every time there is a breaking release which can be often. It is usually better to keep the two version numbers separate as the describe to related but different things.

Also note that CASA uses the date as the soversion (which is a bit overkill in the other direction), so using the major version like this probably breaks it.